sequencer/src/l1_client.rs

```diff
@@ -88,167 +85,56 @@ impl Committable for L1BlockInfo {
-                // skip this one.
-                tracing::warn!("got block from L1 block stream with no number");
-                continue;
-            };
-            let head = head.as_u64();
-
-            // Take a read lock on the snapshot to check if it needs to be updated.
-            let snapshot = snapshot.upgradable_read().await;
-            if head <= snapshot.head {
-                // We got a notification for a new block which is not newer than the current
-                // snapshotted head. This can happen due to an L1 reorg. We are not allowed to
-                // propose an L1 block number which is older than that of the previous Espresso
-                // block, so we skip this notification.
-                //
-                // This does mean that we may end up proposing with a block number that is newer
-                // than the current L1 head, if the height of the L1 is reduced in a reorg. This is
-                // not the best for UX, as rollups will have to block until the new L1 block is
-                // produced, but the L1 chain will _eventually_ (and usually quickly) catch up, and
-                // this only happens in very rare cases anyways. In such rare cases we prioritize
-                // maintaining our invariants (increasing L1 block numbers) over UX.
-                tracing::warn!(
-                    "got L1 head {head} which is not newer than previous head {}",
-                    snapshot.head
-                );
-                continue;
-            }
-
-            // A new block has been produced. This happens fairly rarely, so it is now ok to poll to
-            // see if a new block has been finalized.
-            let finalized = match get_finalized_block(&rpc, finalized_tag).await {
-                Ok(finalized) => Some(finalized),
-                Err(err) => {
-                    tracing::error!("failed to get finalized block: {err}");
-
-                    // If we cannot get the finalized block for any reason, don't let this stop us
-                    // from updating the L1 head. By returning `None` here, we will proceed to
-                    // update the head but not the finalized block, and then we will reset the
-                    // connection, in the hopes that this fixes whatever went wrong with
-                    // `get_finalized_block`.
-                    None
-                }
-            };
-
-            // Update the snapshot.
-            let mut snapshot = RwLockUpgradableReadGuard::upgrade(snapshot).await;
-            snapshot.head = head;
-            if let Some(finalized) = finalized {
-                snapshot.finalized = finalized;
+    /// Proxy to `Provider.get_block_number`.
+    async fn get_block_number(&self) -> u64 {
+        loop {
+            if let Ok(n) = self.agent().get_block_number().await {
+                return n.as_u64();
+            } else {
+                sleep(self.retry_delay).await;
```

Comment:
We should log when errors happen here

Reply:
fixed in dd5425e8e9a1cfb897a0fe2b374dacec06ef7498